### PR TITLE
Replace usage of `apoc.do.when` by sub-queries

### PR DESCRIPTION
**Pull request by Andy2003**
# Description

Replace usage of `apoc.do.when` by sub-queries

## Complexity

Complexity: Medium

# Checklist

- [ ] Documentation has been updated
- [X] TCK tests have been updated
- [ ] Integration tests have been updated
- [ ] Example applications have been updated
- [ ] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2752.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by darrellwarde** [comment]
Thanks for this, we love any opportunity to reduce our use of Apoc! I've added a performance test for this in #2754 - I'll get that merged into `dev` and then we can see if there's a performance impact for this!

File: packages/graphql/tests/tck/directives/auth/arguments/bind/interface-relationships/implementation-bind.test.ts [hunk 1]
```diff
@@ -264,18 +264,13 @@ describe("Cypher Auth Allow", () => {
             CALL {
             	 WITH this
             WITH this
-            OPTIONAL MATCH (this)-[this_has_content0_relationship:HAS_CONTENT]->(this_content0:Comment)
+            MATCH (this)-[this_has_content0_relationship:HAS_CONTENT]->(this_content0:Comment)
             WHERE this_content0.id = $updateUsers_args_update_content0_where_Commentparam0
-            CALL apoc.do.when(this_content0 IS NOT NULL, \\"
             WITH this, this_content0
```
Comment (Andy2003) [inline packages/graphql/tests/tck/directives/auth/arguments/bind/interface-relationships/implementation-bind.test.ts:269]: This block must be a subquery:

https://github.com/neo4j/graphql/pull/2752/files#diff-115f271fbaa3e70c8083dd3fde505e1dee6d583f737e5bf58225bfd1e367a3e5R269-R273

File: packages/graphql/src/translate/create-update-and-params.ts [hunk 2]
```diff
@@ -273,75 +283,24 @@ export default function createUpdateAndParams({
...
+                            innerUpdate.push(`RETURN collect(${META_CYPHER_VARIABLE}) as update_meta`);
+                            returnMetaStatement = `meta AS update${idx}_meta`;
+                            intermediateWithMetaStatements.push(`WITH *, update${idx}_meta AS meta`);
+                        }
 
-                            if (context.subscriptionsEnabled) {
-                                returnMetaStatement = `meta AS update${idx}_meta`;
-                                intermediateWithMetaStatements.push(`WITH *, update${idx}_meta AS meta`);
-                            }
+                        if (context.subscriptionsEnabled) {
```
Comment (angrykoala) [inline packages/graphql/src/translate/create-update-and-params.ts:297]: Could these 2 `if(context.subscriptionsEnabled)` statements merged into one?

**Review by angrykoala — COMMENTED** [review]
Great changes here, I juest have a couple of small questions

**Comment by Andy2003** [comment]
@angrykoala can you rerun the "UI Build", please? It did fail b/c of an error while uploading files

**Review by angrykoala — APPROVED** [review]

**Pull request merged**